### ISSUE TITLE
Why use ResNet18 at SimCLR step?

### ISSUE THREAD

**Issue by Jie2World** [wvangansbeke/Unsupervised-Classification#30]
SCAN is a surprising work. But I keep wonder why select ResNet18.  Google-research offer a standard ResNet50 model for SimCLR.
What of the effect on clustering if i use the standard ResNet50 to replace ResNet18?

**Comment by wvangansbeke** [comment]
Hi @Jie2World,

Indeed, a deeper network will boost the performance. Changing network backbones should be straightforward with this codebase. Feel free to try it out.

There are multiple reasons why we used a ResNet18 actually:
- __Google uses Tensorflow__: So we would have to transfer the weights from Tensorflow to Pytorch. I'm not even sure their weights are available for other datasets than ImageNet. (Btw, keep in mind that we use a ResNet50 for ImageNet provided by MoCo.)

- __The memory consumption__: Notice that the original SimCLR implementation from Google does not use a memory bank. So, large batch sizes are required during training which results in more GPU memory usage. I unfortunately don't have the resources from Google. So it is more compelling to adopt a ResNet18 in my case.

- __ResNet18 is much faster__ to train and test: If the method works with a ResNet18, it is highly likely going to work with a ResNet50.

- __To have fair comparisons__ with prior work: We didn't want to use a deeper network than a ResNet34 (for the smaller datasets) since a ResNet34 is used in prior work. A ResNet34 is likely going the boost our results even more though. (We also mention this in the paper.) Therefore, adopting a ResNet50 seems unfair to me.

- __Timeline__: We more or less finished the paper before SimCLR came out and were using the implementation from Wu et al. back then. They also used a ResNet18 for the smaller datasets. 

- __Focus__: For the paper we simply wanted to give an indication which pretext tasks are better-suited for the task of semantic clustering. To achieve this, we experimented with other tasks than instance discrimination. We always kept the backbone fixed for different pretext tasks. Changing the backbone seems orthogonal to our work, since the method, described in the paper, does not rely on a specific architecture. 

Hope this helps.
